python/hsfs/core/tfdata_engine.py

```diff
@@ -320,3 +425,40 @@ def _get_s3_dataset_files(training_dataset_location, split):
             input_files.append(path + "/" + s3_obj_summary.key)
 
     return input_files
+
+
+def _convert2tfdtype(input_type):
+    supported_types = {
+        "string": tf.string,
+        "short": tf.int16,
+        "int": tf.int32,
+        "long": tf.int32,
+        "float": tf.float32,
+        "double": tf.float32,
+    }
+
+    try:
+        tf_type = supported_types[input_type]
+    except KeyError:
+        raise ValueError("Unknown type of value, please report to hsfs maintainers")
+    return tf_type
+
+
+def _convert2float32(input):
+    if input.dtype == tf.string:
+        raise ValueError(
+            "tf.string feature is not allowed here. please provide process=False and preprocess "
+            "dataset accordingly"
+        )
+    elif input.dtype in [
```

Comment:
can you define the list of types at the class level in a constant, so we have one central place to change them.

Reply:
if the `input` object has these types also as python types, can we use `isinstance()`

Reply:
I thought the `input` variable here is a tfrecord example. and I am not sure how they implemented it, but if the python type of this matches the dtype attribute, you could use `isinstance`